File: lista de exercicios/questão 11/questao.go
package main

import "fmt"

// Interface que define o método calcularSalario
type Funcionario interface {
    CalcularSalario() float64
}

// Struct para funcionários horistas
type FuncionarioHorista struct {
    Horas         int
    ValorPorHora  float64
}

// Implementação do método CalcularSalario para FuncionarioHorista
func (f FuncionarioHorista) CalcularSalario() float64 {
    return float64(f.Horas) * f.ValorPorHora
}

// Struct para funcionários assalariados
type FuncionarioAssalariado struct {
    SalarioFixo float64
}

// Implementação do método CalcularSalario para FuncionarioAssalariado
func (f FuncionarioAssalariado) CalcularSalario() float64 {
    return f.SalarioFixo
}

func main() {
    horista := FuncionarioHorista{Horas: 160, ValorPorHora: 50}
    assalariado := FuncionarioAssalariado{SalarioFixo: 3000}

    fmt.Printf("Salário do horista: R$%.2f\n", horista.CalcularSalario())
    fmt.Printf("Salário do assalariado: R$%.2f\n", assalariado.CalcularSalario())
}

File: lista de exercicios/questão 11/questao.py
from abc import ABC, abstractmethod

class Funcionario(ABC):
    @abstractmethod
    def calcular_salario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, horas, valor_por_hora):
        self.horas = horas
        self.valor_por_hora = valor_por_hora

    def calcular_salario(self):
        return self.horas * self.valor_por_hora

class FuncionarioAssalariado(Funcionario):
    def __init__(self, salario_fixo):
        self.salario_fixo = salario_fixo

    def calcular_salario(self):
        return self.salario_fixo

# Exemplo de uso
horista = FuncionarioHorista(160, 50)
assalariado = FuncionarioAssalariado(3000)

print(f"Salário do horista: R${horista.calcular_salario()}")
print(f"Salário do assalariado: R${assalariado.calcular_salario()}")
